### PR TITLE
fix(app): fix connect robot via ip address functionality and style

### PR DESCRIPTION
**Pull request by koji**
# Overview
This will fix the timing of `startDiscovery`(#10593) and conditional rendering of the refresh and try again button(#10594). In addition, this PR will fix the styling feedback that is pointed out in #10618

# Changelog
- Add useEffect to run `startDiscovery` when a user opens the slideout
- Fix conditional rendering part for the blue link button 
- Remove unnecessary expect() from test code (This test code was my first test cases and used expect() for text display tests)

# Review requests
- When opening the slideout, startDiscovery starts. (When the app starts, `startDiscovery` is called. To check this, please open the slideout  30s after starting the app
- When adding `localhost` or robot IP address, the status `available` and the slideout `refresh` button
- When adding a non-existing IP address or hostname, the status `not found` and the slideout shows `try again` button.

After checking this, please make sure that the slideout displays `localhost`. If you remove `localhost` from the slideout, the app won't recognize your dev robot-server.

# Risk assessment
- low

### CODE REVIEW

File: app/src/organisms/AppSettings/ConnectRobotSlideout.tsx [hunk 1]
```diff
@@ -100,27 +102,40 @@ export function ConnectRobotSlideout(
...
+                  {displayLinkButton(t('try_again'))}
                 </>
+              ) : (
+                displayLinkButton(t('shared:refresh'))
               ),
             ]
           )}
         </Flex>
-        <IpHostnameList mostRecentAddition={mostRecentAddition} />
+        <IpHostnameList
```
Comment (jerader) [inline app/src/organisms/AppSettings/ConnectRobotSlideout.tsx]: there's a bit of a weird formatting issue here, the Ip hostname list has a slight left margin compared to the rest of the text in the slideout

Screen Shot 2022-06-16 at 11 58 18 AM

File: app/src/organisms/AppSettings/ManualIpHostnameForm.tsx [hunk 2]
```diff
@@ -33,14 +29,43 @@ const FlexForm = styled.form`
 `
 
 const StyledInput = styled.input`
-  height: ${SPACING.spacing5};
   width: 100%;
   flex: 6;
   margin: ${SPACING.spacing2} 0;
-  border: 1px solid ${COLORS.medGrey};
+  background-color: ${COLORS.white};
```
Comment (jerader) [inline app/src/organisms/AppSettings/ManualIpHostnameForm.tsx:35]: maybe there should be a padding around where you enter the text so the cursor isn't at the very end. Also, how come you didn't use the `InputField` component in the Apps folder instead?

Screen Shot 2022-06-16 at 12 02 30 PM

File: app/src/organisms/AppSettings/__tests__/ConnectRobotSlideout.test.tsx [hunk 3]
```diff
@@ -162,8 +143,8 @@ describe('ConnectRobotSlideout', () => {
       })
       await fireEvent.click(addButton)
     })
-    expect(getByText(notFoundIpAddress)).toBeInTheDocument()
-    expect(getByText('Not Found')).toBeInTheDocument()
+    findByText(notFoundIpAddress)
+    findByText('Not Found')
```
Comment (jerader) [inline app/src/organisms/AppSettings/__tests__/ConnectRobotSlideout.test.tsx:149]: this is cool, I have never used `findByText` before!

**Review by jerader — COMMENTED** [review]
functionality seems good to me, I just left a few comments about the style.

File: app/src/organisms/AppSettings/__tests__/ConnectRobotSlideout.test.tsx (hunk 3 above)
Comment (koji) [reply app/src/organisms/AppSettings/__tests__/ConnectRobotSlideout.test.tsx:149]: I think when we need to `async` for a test case, `findByText` will be good.

File: app/src/organisms/AppSettings/ManualIpHostnameForm.tsx (hunk 2 above)
Comment (koji) [reply app/src/organisms/AppSettings/ManualIpHostnameForm.tsx:35]: 5.0.2 is using `formik` and Brian recommended to keep using that for this slidout.

File: app/src/organisms/AppSettings/ManualIpHostnameForm.tsx [hunk 4]
```diff
@@ -54,18 +79,21 @@ export function ManualIpHostnameForm(props: Props): JSX.Element {
     onSubmit: (values, { resetForm }) => {
       const ip = values.ip.trim()
       const inputForm = document.getElementById('ip')
-      if (inputForm) inputForm.style.border = `1px solid ${COLORS.medGrey}`
+      if (inputForm != null)
+        inputForm.style.border = `1px solid ${COLORS.medGrey}`
       addManualIpAndHostname(ip)
-      props.setMostRecentAddition(ip)
-      resetForm({ values: undefined })
+      // setMostRecentAddition(ip)
```
Comment (jerader) [inline app/src/organisms/AppSettings/ManualIpHostnameForm.tsx]: left over comment here

**Review by jerader — APPROVED** [review]
you forgot to delete a comment and looks like you have some merge conflicts. But otherwise lgtm!

**Comment by mmencarelli** [comment]
**Round 2 Feedback on 'Connect a Robot via IP Address' slideout**

[Link to view annotated figma frame ](https://www.figma.com/file/bVtCogdwiXj3ndpk4AWXCx/Milestone-1%3A-Read-and-Execute?node-id=14680%3A290192)


- [ ] (1) Increase button size to 24px vs the 20px it's at 
- [ ] (2) Change this font size to our 'label' styling. So it should be 11px vs the 13px it's at 
- [ ] (3) Align the various states for our text field with the design system (image of side by side comparison outlining the differences included in this PR and in figma if you want to go into the designs). Hover, activated, error and focus-visible need some attention
- [ ] (4) Let's have 8px margin-left on the actual input text so it's not so close to the left hand side 
- [ ] (5) Let's remove the margin-top: 4px on the error text actually. 
- [ ] (6) Are we able to get this text to be a countdown? If it's not in scope, then static text is fine. If that's the case, then please confirm that we're capturing that future change for a later release. Don't want it to fall between the cracks. Also, can we capitalize "searching" so it's "Searching"
- [ ] (7) Update colour of spinner to darkGreyEnabled
- [ ] (8) Can we change this text string to "Not Found" 
- [ ] (9) Can we reverse the order of display here, so any available connections automatically appear at the top? The logic on top of that is that it's in order of user search. 
- [ ] (10) Can we add margin: 4px 0px; to `.mWBlO`
- [ ] (11) Missing hover state on the "refresh" button
- [ ] (12) remove the box-shadow from the button
- [ ] (13) increase font size on the button to 11px (currently around 9px) 


Screen Shot 2022-06-23 at 2 57 36 PM
Screen Shot 2022-06-23 at 2 57 46 PM

**Review by mmencarelli — CHANGES_REQUESTED** [review]
changes requested above!

**Pull request merged**